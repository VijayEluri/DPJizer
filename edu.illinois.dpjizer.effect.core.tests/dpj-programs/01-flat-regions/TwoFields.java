/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package flatregion.twofield;

import java.util.Date;

/**
 * 
 * @author Mohsen Vakilian
 * @author Danny Dig
 *
 */
class TwoFields {
	region r1, r2;
	Date f1 in r1;
	Date f2 in r2;

	TwoFields() {}
	
	void m1(Date x) /*reads r2, writes r1*/ {
		m2(x);
	}
	
	Date m2(Date x) /*reads r2, writes r1*/ {
		f1 = x;
		return f2;
	}

}
