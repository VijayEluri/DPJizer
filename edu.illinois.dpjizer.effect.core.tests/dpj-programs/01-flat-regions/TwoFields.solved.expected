flatregion.twofield.TwoFields.TwoFields() pure
flatregion.twofield.TwoFields.m1(java.util.Date) reads Root : flatregion.twofield.TwoFields.r2 writes Root : flatregion.twofield.TwoFields.r1
flatregion.twofield.TwoFields.m2(java.util.Date) reads Root : flatregion.twofield.TwoFields.r2 writes Root : flatregion.twofield.TwoFields.r1
